### PR TITLE
Bugfix/df 574 whitespace validation

### PR DESCRIPTION
**Pull request by fabiangotzen**

### CODE REVIEW

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.ts [hunk 1]
```diff
@@ -69,11 +69,13 @@ export class ContactComponent implements OnInit {
       ]],
       firstName: [firstName, [
         Validators.required,
-        Validators.maxLength(this.FIRST_NAME_MAX_LENGTH)
+        Validators.maxLength(this.FIRST_NAME_MAX_LENGTH),
+        patternInputValidator(NO_WHITESPACE_REGEX)
```
Comment (birom) [inline alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.ts]: I would prefer a different solution like this eg.: [trim-value-accessor.ts](https://github.com/khashayar/ng-trim-value-accessor/blob/master/src/trim-value-accessor.ts). This trims the input field value.

Comment (fabiangotzen) [reply alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.ts]: I've started out with a different approach in commit https://github.com/alv-ch/alv-portal/pull/270/commits/7ae79b21b511bd463e47a744a9823b56cb6d0a58 as well. Might also be an option.

File: alv-portal-ui/src/app/shared/forms/input/input-field/input-field.component.ts [hunk 2]
```diff
@@ -131,6 +131,15 @@ export class InputFieldComponent extends AbstractInput implements AfterViewInit
     this.input.emit(event);
   }
 
+  onBlur() {
+    if (this.type === 'text' && this.control.value) {
+      const trimmedValue = this.control.value.trim();
+      if (trimmedValue.length === 0) {
```
Comment (birom) [inline alv-portal-ui/src/app/shared/forms/input/input-field/input-field.component.ts]: Why do we check the length? This does not really work with eg. regex validator.
selection_155

Comment (fabiangotzen) [reply alv-portal-ui/src/app/shared/forms/input/input-field/input-field.component.ts]: If I put only spaces into a text field with the length check:
image

If the length check is omitted and again, I only put spaces in:
image

I'll have a look at a solution if combined with other validators.

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.spec.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { SharedModule } from '../../../../shared/shared.module';
 import { FormGroup } from '@angular/forms';
 import { ContactComponent } from './contact.component';
 import { emptyContactFormValue } from './contact-form-value.types';
+import { By } from 'protractor';
```
Comment (ganqqwerty) [inline alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.spec.ts]: huh?

Comment (fabiangotzen) [reply alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/contact/contact.component.spec.ts]: woops, fixed

**Pull request merged**